internal/binarylog/binarylog.go

```diff
@@ -21,38 +21,64 @@
 package binarylog
 
 import (
+	"context"
 	"fmt"
 	"os"
 
 	"google.golang.org/grpc/grpclog"
 	"google.golang.org/grpc/internal/grpcutil"
 )
 
-// Logger is the global binary logger. It can be used to get binary logger for
-// each method.
+var grpclogLogger = grpclog.Component("binarylog")
+
+// Logger specifies MethodLoggers for method names with a Log call that don't
+// take a context.
 type Logger interface {
 	GetMethodLogger(methodName string) MethodLogger
 }
 
+// LoggerContext specifies MethodLoggers for method names with a Log call that
+// takes a context.
+type LoggerContext interface {
+	GetMethodLoggerContext(methodName string) MethodLoggerContext
+}
+
+type wrappedLogger struct {
+	logger Logger
+}
+
+func (wl *wrappedLogger) GetMethodLoggerContext(methodName string) MethodLoggerContext {
+	if wl.logger == nil {
+		return nil
+	}
```

Comment:
Can this ever happen?  If so is there no way to prevent it?

Reply:
No, but I added it for defensive. I guess since we control this internally, we can get rid of this branch and just assume that we actually populate it.